chaos-days/blog/2024-10-14-realistic-benchmarks/index.md

```diff
@@ -0,0 +1,226 @@
+tasks second to be significantly reduced in resources in relation to our 
+smaller clusters ([G3-S HA Plan](https://accounts.cloud.dev.ultrawombat.com/consoleadmin/clusterplans/0af08654-28ec-413a-8dcf-c5938e828ddd)) since 
+these can process 50-70 tasks per second. 
+
+We also expect that we can scale this base configuration linearly, and that 
+the processing tasks rate to grow initially a bit faster than linearly due to 
+the lower relative overhead, and if we keep expanding further to flatten due 
+to the partition count being a bottleneck.
+
+### Actual
+
+#### Benchmarking a realistic process model
+
+In our previous benchmarks we used a simple process model with a single 
+task with several decision symbols. For the newer benchmarks we wanted to 
+increase significantly the number of symbols and service tasks used as well 
+to make the load more configurable.
+
+![bank-customer-complaint-dispute-handling](bank-customer-complaint-dispute-handling.png)
+
+For this realistic benchmark, we tried to use a situation closer to a real 
+world scenario, where a bank customer complaint dispute handling process is
+handled by several service tasks. Additionally, fraud claim 
+investigation part of the process is unfolded in multiple instances. This 
+can be configured in our instance payload by how many disputes we send to 
+scale up or down the number of multi instances in the fraud claim 
+investigation, which enables to scale the load of each process instance.
+
+Finally, in deploying our benchmark we can define the number of instances 
+started per second. After choosing a uniform payload for all tests we can 
+start benchmarking or different cluster configurations where we measure the 
+performance by how many process instances are completed per second, while 
+maintaining low backpressure (bellow 10%) as to preserve user experience. 
+
+#### Minimal Viable Requirements for our Cluster
+
+For our minimal viable cluster configuration we determined though several 
+metrics on users and our own experience that we need to be able to handle 
+at least 40 process instances per second, or 3.5 million tasks per day.
+
+Other metrics that we want to preserve and keep track are the backpressure 
+to preserve user experience, guarantee that exporting speed can keep up 
+with the processing speed, write to import latency which tells us how long 
+it takes for a record to be written to being imported by our other 
+apps such as the operator.
+
+#### Reverse Engineering the Cluster Configuration
+
+For our new configurations the only resources that we are going to change 
+are the ones relevant to the factors described above. These are the 
+resources allocated to our zeebe-brokers, gateway and elasticSearch.
+
+Our starting point in resources was the configuration for our [G3-S HA Plan](https://accounts.cloud.dev.ultrawombat.com/consoleadmin/clusterplans/0af08654-28ec-413a-8dcf-c5938e828ddd) 
```

Comment:
:x: Again, don't think people can access this link. We already show it above anyway, no need to mention it again.